sklearn/tests/test_init.py

```diff
@@ -18,3 +20,17 @@ def test_import_skl():
     # "import *" is discouraged outside of the module level, hence we
     # rely on setting up the variable above
     assert_equal(_top_import_error, None)
+
+
+def test_import_sklearn_no_warnings():
+    # Test that importing scikit-learn doesn't raise any warnings.
+
+    message = subprocess.check_output(['python', '-Wdefault',
+                                       '-c', 'import sklearn'],
```

Comment:
Not sure there is a better way of testing this? Trying to run the import in a separate process with `mupltiprocessing` but I couldn't see the deprecation warning. Using `importlib.reload` + "always" warning filter also didn't work. Besides I wasn't sure how it might interact with pytest warning capture, so using `subprocess` seems easier to isolate environment.

Reply:
Does this really fail at master? Surely `sklearn/__init__.py` is light enough that no warning is triggered...?

Reply:
It does fail on master with Python 3.7 without this PR as described in the original issue https://github.com/scikit-learn/scikit-learn/issues/11121#issue-325697069 

`sklearn/__init__.py` imports `base.py` which imports `sklearn/utils/__init__.py` which imports `collections.abc.Sequence`. There is second place it is raised from the `__init__` I think as well.

Reply:
I don't really see the point in testing the import of sklearn, and not testing the import of every module for either an internal DeprecationWarning or no warning.